clouddriver-dcos/src/test/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerLbIdSpec.groovy

```diff
@@ -0,0 +1,85 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import spock.lang.Specification
+
+class DcosSpinnakerLbIdSpec extends Specification {
+    static final def ACCOUNT = "spinnaker"
+    static final def REGION = "us-west-1"
+    static final def LOAD_BALANCER = "loadbalancer"
+    static final def INVALID_ACCOUNT = "invalid-account"
+    static final def INVALID_MARATHON_PART = "-iNv.aLiD-"
+
+    void "static factory method should return an empty optional if the marathon app id is invalid"() {
+        expect:
+            def dcosPath = DcosSpinnakerLbId.parseVerbose(appId, account)
```

Comment:
Since you're using `ACCOUNT` for each test case, could just pass it to `parseVerbose()` directly.
For the remaining set of `appId` values, you could use something like:
```
where:
  appId << ["/", "", "/${ACCOUNT}", "${ACCOUNT", ///]
```